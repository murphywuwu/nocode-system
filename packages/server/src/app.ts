import Koa from 'koa';
import compose from 'koa-compose';

import middlewares from './middlewares';

const app = new Koa();

app.use(compose(middlewares));

export default app;
