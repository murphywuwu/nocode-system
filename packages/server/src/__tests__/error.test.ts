import supertest from 'supertest';

import app from '../app';

describe('error handle', function () {
  let request;
  beforeAll(function () {
    request = supertest(app.callback());
  });

  afterAll(function () {
    request = null;
  });

  test('hello world works', async () => {
    const response = await request.get('/hello');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();

    // code
    // msg
    // data
  });
});
