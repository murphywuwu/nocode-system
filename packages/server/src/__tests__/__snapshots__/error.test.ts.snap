// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error handle hello world works 1`] = `"Hello World"`;
